### ISSUE TITLE
Issues of training dreambooth using DoRA with text encoder

### ISSUE THREAD

**Issue by rfan-debug** [huggingface/diffusers#7431]
### Describe the bug

I executed the following command to train a dreambooth model using the example script from the document. The only change I made is to change the `--train_text_encoder_ti` to `--train_text_encoder`. However, the loss is **always NaN**. 

References:
+ https://github.com/huggingface/diffusers/blob/main/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

I inspected intermediate output during training, i noticed that the output from the second text encoder outputs `NaN` when you set `--use_dora` and `--train_text_encoder` at the same time. Meanwhile, the output of the first text encoder seems to be normal under the same setting.

The second text encoder is `CLIPTextModelWithProjection`. My guess is that somehow this text encoder class is not fully compatible with DoRA in a forward pass under the training mode (aka. when `.train()` is called)? However, I cannot figure out the root cause. Asking here to see if anyone else ran into the same problem. 

### Reproduction

Run this command with an nvidia GPU

```shell
accelerate launch train_dreambooth_lora_sdxl_advanced.py \
  --pretrained_model_name_or_path="playgroundai/playground-v2.5-1024px-aesthetic"  \
  --dataset_name="linoyts/3d_icon" \
  --instance_prompt="3d icon in the style of TOK" \
  --validation_prompt="a TOK icon of an astronaut riding a horse, in the style of TOK" \
  --output_dir="3d-icon-SDXL-LoRA" \
  --do_edm_style_training \
  --caption_column="prompt" \
  --mixed_precision="fp16" \
  --resolution=1024 \
  --train_batch_size=3 \
  --repeats=1 \
  --gradient_accumulation_steps=1 \
  --gradient_checkpointing \
  --learning_rate=1.0 \
  --text_encoder_lr=1.0 \
  --optimizer="prodigy"\
  --train_text_encoder \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --rank=8 \
  --max_train_steps=1000 \
  --checkpointing_steps=2000 \
  --seed="0" \
  --use_dora
```

### Logs

```shell
/opt/conda/envs/lora-train/lib/python3.10/site-packages/accelerate/accelerator.py:394: UserWarning: `log_with=tensorboard` was passed but no supported trackers are currently installed.
  warnings.warn(f"`log_with={log_with}` was passed but no supported trackers are currently installed.")
03/22/2024 05:58:27 - INFO - __main__ - Distributed environment: NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: fp16

You are using a model of type clip_text_model to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
You are using a model of type clip_text_model to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
03/22/2024 05:58:28 - INFO - __main__ - Performing EDM-style training!
03/22/2024 05:58:40 - WARNING - __main__ - Learning rates were provided both for the unet and the text encoder- e.g. text_encoder_lr: 1.0 and learning_rate: 1.0. When using prodigy only learning_rate is used as the initial learning rate.
Using decoupled weight decay
03/22/2024 05:58:40 - INFO - datasets - PyTorch version 2.1.0+cu118 available.
Resolving data files: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 24/24 [00:00<00:00, 62.64it/s]
/opt/conda/envs/lora-train/lib/python3.10/site-packages/PIL/Image.py:3186: DecompressionBombWarning: Image size (122880000 pixels) exceeds limit of 89478485 pixels, could be decompression bomb DOS attack.
  warnings.warn(
/opt/conda/envs/lora-train/lib/python3.10/site-packages/PIL/Image.py:3186: DecompressionBombWarning: Image size (132710400 pixels) exceeds limit of 89478485 pixels, could be decompression bomb DOS attack.
  warnings.warn(
03/22/2024 05:58:53 - INFO - __main__ - ***** Running training *****
03/22/2024 05:58:53 - INFO - __main__ -   Num examples = 23
03/22/2024 05:58:53 - INFO - __main__ -   Num batches each epoch = 8
03/22/2024 05:58:53 - INFO - __main__ -   Num Epochs = 125
03/22/2024 05:58:53 - INFO - __main__ -   Instantaneous batch size per device = 3
03/22/2024 05:58:53 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 3
03/22/2024 05:58:53 - INFO - __main__ -   Gradient Accumulation steps = 1
03/22/2024 05:58:53 - INFO - __main__ -   Total optimization steps = 1000
Steps:   0%|▏                                                                                                                | 2/1000 [00:10<1:29:37,  5.39s/it, loss=nan, lr=1]
```

### System Info

peft==0.10.0
transformers==4.39.0
git+https://github.com/huggingface/diffusers.git@3e1097cb63c724f5f063704df5ede8a18f472f29
torch==2.1.0+cu118
torchaudio==2.1.0+cu118
torchvision==0.16.0+cu118

### Who can help?

Maybe the author of https://github.com/huggingface/diffusers/blob/main/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py could help?

**Comment by sayakpaul** [comment]
Cc: @linoytsaban

**Comment by linoytsaban** [comment]
Hey @rfan-debug! I see you're using EDM style training, did you encounter this issue also when training a DoRA with text-encoder training?

**Comment by rfan-debug** [comment]
Hey, @linoytsaban . I also tried to remove the `--do_edm_style_training` option from the script but still got the same `NaN` loss. Also, i don't think that option matters in this base model because the base model we used here uses a EDM scheduler: https://huggingface.co/playgroundai/playground-v2.5-1024px-aesthetic/blob/main/scheduler/scheduler_config.json#L2

**Comment by asomoza** [comment]
I was testing this and I can reproduce this issue, this happens when training with `use_dora` and with the second text encoder `CLIPTextModelWithProjection`.

This trains succesfully:

```python
    if args.train_text_encoder:
        text_lora_config = LoraConfig(
            r=args.rank,
            use_dora=args.use_dora,
            lora_alpha=args.rank,
            init_lora_weights="gaussian",
            target_modules=["q_proj", "k_proj", "v_proj", "out_proj"],
        )
        text_lora_config_two = LoraConfig(
            r=args.rank,
            # use_dora=args.use_dora,
            lora_alpha=args.rank,
            init_lora_weights="gaussian",
            target_modules=["q_proj", "k_proj", "v_proj", "out_proj"],
        )
        text_encoder_one.add_adapter(text_lora_config)
        text_encoder_two.add_adapter(text_lora_config_two)
```

But this not (NaN loss)
```python
    if args.train_text_encoder:
        text_lora_config = LoraConfig(
            r=args.rank,
            # use_dora=args.use_dora,
            lora_alpha=args.rank,
            init_lora_weights="gaussian",
            target_modules=["q_proj", "k_proj", "v_proj", "out_proj"],
        )
        text_lora_config_two = LoraConfig(
            r=args.rank,
            use_dora=args.use_dora,
            lora_alpha=args.rank,
            init_lora_weights="gaussian",
            target_modules=["q_proj", "k_proj", "v_proj", "out_proj"],
        )
        text_encoder_one.add_adapter(text_lora_config)
        text_encoder_two.add_adapter(text_lora_config_two)
```

**Comment by sayakpaul** [comment]
Could it be because it needs some learning rate adjustments?

**Comment by asomoza** [comment]
Tried with really high or low learning rates, also changed to other schedulers, training fails with all of them.

**Comment by sayakpaul** [comment]
Is this still a problem?

**Comment by sayakpaul** [comment]
Closing due to inactivity.
